### PR TITLE
Add support for Windows

### PR DESCRIPTION
**Pull request by michaelkschmidt**
For reasons I don't understand, System.cmd("which",[]) just hangs on Windows

### CODE REVIEW

File: lib/mix/tasks/firmware.burn.ex [hunk 1]
```diff
@@ -75,6 +75,8 @@ defmodule Mix.Tasks.Firmware.Burn do
            ask_pass = System.get_env("SUDO_ASKPASS") || "/usr/bin/ssh-askpass"
            System.put_env("SUDO_ASKPASS", ask_pass)
            {"sudo", ["fwup"] ++ args}
+        {_, :nt} ->
+           {"fwup", args}
```
Comment (fhunleth-lkc) [inline lib/mix/tasks/firmware.burn.ex]: Is `fwup` working fine on Windows? I assume that you're using 0.9.0-dev. I'm planning on making an official `fwup` release really soon, but if there are any annoyances, now would be a good time for me to fix them.

Comment (michaelkschmidt) [inline lib/mix/tasks/firmware.burn.ex]: It has been working great!  I cross compiled it from a Linux VM, and haven't had a single issue.  It even detects the SD Card correctly :+1:

Comment (fhunleth-lkc) [inline lib/mix/tasks/firmware.burn.ex]: Super! @GregMefford wrote the SD Card detection code and will be thrilled to hear.

**Comment by fhunleth** [comment]
Since I don't think that any of us said this yet, thank you so much for the Windows patches. We're in the process of catching up from ElixirConf, but we'll get to this patch soon!

Also, I just officially released fwup v0.9.0 which has Windows support so we're getting closer.

**Comment by michaelkschmidt** [comment]
No problem.  When you get back to the patches, let me know if there is something you would prefer changed.  I tried hard to get tar to extract the .tar.xz files, but I kept running into weird interactions with the MinGW/Cygwin path.  7zip is a bit of a punt, but at least its easily installed via Chocolatey.

Speaking of Chocolatey, do you plan to distribute fwup on it?  I would be glad to try to help.  The squashfs tools should probably be put on there as well, too, but I'm not sure who would be the appropriate person to do so.

**Comment by fhunleth** [comment]
Adding fwup and the squashfs tools to Chocolatey would be ideal. I've never submitted anything to Chocolatey before, but now that you mention it, I'd like to do it. I'm hanging it up for the night, but let's chat over Slack on this and see if we can get these included.

File: lib/mix/tasks/firmware.burn.ex (hunk 1 above)
Comment (GregMefford) [inline lib/mix/tasks/firmware.burn.ex]: Very exciting when things just work!

**Comment by mobileoverlord** [comment]
This is great, thanks @michaelkschmidt I am going to merge this in and add it to the new pkg branch coming in.

Thanks! :bowtie:

**Comment by fhunleth** [comment]
@michaelkschmidt Maybe I haven't seen it yet, but how did you handle toolchains on Windows? Did you build your own?

**Comment by michaelkschmidt** [comment]
The short answer is I built my own toolchain.

I was hoping to get your feedback on the long answer--I'll ping you on slack

**Pull request merged**